Error in generic selection for shmem_g

## Background information

**Problem**: There is essentially a typo in the generic selection expression for [`shmem_g` in `shmem.h.in`](https://github.com/open-mpi/ompi/blob/master/oshmem/include/shmem.h.in#L300). The macro for `shmem_g` accepts the arguments `addr` and `pe`, however, the controlling expression for the generic selection uses `dst`.

**Resolution**: Replace `dst` with `addr` on [line 301 of `shmem.h.in`](https://github.com/open-mpi/ompi/blob/master/oshmem/include/shmem.h.in#L301).

### What version of Open MPI are you using?

v2.1.1

### Describe how Open MPI was installed
From source tarball

### Please describe the system on which you are running

* Operating system/version: Ubuntu 16.04 LTS
* Computer hardware: MacBook Pro

-----------------------------

## Details of the problem

The OpenSHMEM Specification example codes that use `shmem_g` fail to compile with OSHMEM. For example, see [shmem_g_example.c](https://github.com/openshmem-org/specification/blob/osh_spec_next/example_code/shmem_g_example.c).

```shell
$ oshcc -Wall -Wextra -o shmem_g_example.cx shmem_g_example.c
In file included from shmem_g_example.c:2:0:
shmem_g_example.c: In function ‘main’:
/opt/oshmem/2.1.1/include/shmem.h:302:17: error: ‘dst’ undeclared (first use in this function)
     _Generic(&*(dst),                                        \
                 ^
shmem_g_example.c:12:11: note: in expansion of macro ‘shmem_g’
       y = shmem_g(&x, npes-1);
           ^
/opt/oshmem/2.1.1/include/shmem.h:302:17: note: each undeclared identifier is reported only once for each function it appears in
     _Generic(&*(dst),                                        \
                 ^
shmem_g_example.c:12:11: note: in expansion of macro ‘shmem_g’
       y = shmem_g(&x, npes-1);
           ^
```

--- a/oshmem/include/shmem.h.in
+++ b/oshmem/include/shmem.h.in
@@ -298,7 +298,7 @@ OSHMEM_DECLSPEC  long long shmem_longlong_g(const long long* addr, int pe);
 OSHMEM_DECLSPEC  long double shmem_longdouble_g(const long double* addr, int pe);
 #if OSHMEM_HAVE_C11
 #define shmem_g(addr, pe)                                    \
-    _Generic(&*(dst),                                        \
+    _Generic(&*(addr),                                        \
             char*:        shmem_char_g,                      \
             short*:       shmem_short_g,                     \
             int*:         shmem_int_g,                       \

